### DISCUSSION TITLE
gitlab cli - intended audience

### DISCUSSION THREAD

**Discussion by nyue** [python-gitlab/python-gitlab#2092] [Q&A]
Hi,

  I am having a look at the gitlab CLI and am not sure if that is intended as a developer tool or an administrator tool.

  I have Maintainer and Owner access to a corporate GitLab instance that is managed by the IT department.

  The gitlab CLI documentation refers to config file and also no mention about login (I see --sudo option) so I am unsure.

  Is there some documentation on how to use gitlab CLI (including the setup).

  I am using the python API but am also looking at the CLI as another option.

Cheers

**Comment by nejch** [comment]
Hi @nyue that's a good question. The CLI needs a bit of work to make it more convenient, but please check this issue/comment for some workaround on how to imitate an interactive login currently: https://github.com/python-gitlab/python-gitlab/issues/960#issuecomment-747345388

The CLI setup docs are here https://python-gitlab.readthedocs.io/en/stable/cli-usage.html
And auto-generated reference for all subcommands here https://python-gitlab.readthedocs.io/en/stable/cli-objects.html

I'd say it's useful for both developers and maintainers/admins, but it's really more for scripting and automation, nothing fancy If you want something more like an interactive experience with all the bells and whistles, I'd recommend https://github.com/profclems/glab instead which is getting more activity recently, is more developer-focused but covers a lot less of the API functionality.

Personally I use the CLI when I need to feed some data from GitLab to another tool and don't want to write entire scripts for it when the CLI can do it in a one-liner. This is useful in CI pipelines for example. It would be interesting to hear your use cases though!

**Comment by nyue** [comment]
Thank you @nejch 

The cli-usage link was really help, it got it going.

One question though, in the example code, it shows the specifying of single IDs, I am interested in the following
[Approval Rule Creation](https://python-gitlab.readthedocs.io/en/stable/cli-objects.html#gitlab-project-approval-rule-create)

It asks for multiple user IDs, how does one provide multiple IDs ?

Cheers
